### PR TITLE
Remove Whitehall asset

### PR DESCRIPTION
**Pull request by yuetylauiris**
`Whitehall` app no longer using legacy url now so we can delete the related whitehall_assets methods/helpers/tests.

We are keeping the media ones since there are still so use cases of that. 

This PR is dependent on [the frontend csv preview PR](https://github.com/alphagov/frontend/pull/3886) as frontend is still using whitehall_asset endpoint.

[Trello card](https://trello.com/c/Kp9nquOr/71-task-remove-whitehallasset-concept-from-asset-manager-and-gds-api-adapters)

### CODE REVIEW

File: lib/gds_api/test_helpers/asset_manager.rb [hunk 1]
```diff
@@ -31,14 +31,9 @@ def stub_asset_manager_has_an_asset(id, atts, filename = "")
           .to_return(body: "Some file content", status: 200)
       end
 
-      def stub_asset_manager_has_a_whitehall_asset(legacy_url_path, atts)
-        response = atts.merge("_response_info" => { "status" => "ok" })
-
-        stub_request(:get, "#{ASSET_MANAGER_ENDPOINT}/whitehall_assets/#{legacy_url_path}")
-          .to_return(body: response.to_json, status: 200)
-
+      def stub_asset_manager_has_a_whitehall_media(legacy_url_path, content)
```
Comment (JonathanHallam) [inline lib/gds_api/test_helpers/asset_manager.rb:34]: minor - should this be `has_a_whitehall_media_asset`?

**Review by JonathanHallam — COMMENTED** [review]
Looks good, minor comment on naming things. Thanks for this! Good to end some Whitehall weirdness!

File: test/asset_manager_test.rb [hunk 2]
```diff
@@ -108,18 +88,12 @@
 
   describe "a Whitehall asset exists" do
     before do
-      stub_asset_manager_has_a_whitehall_asset(
+      stub_asset_manager_has_a_whitehall_media(
```
Comment (JonathanHallam) [inline test/asset_manager_test.rb:91]: Sorry, all of these will need updating too with the rename

Comment (yuetylauiris) [reply test/asset_manager_test.rb:91]: of cos the one repo I forgot to check 💯 Thanks for reminding me!

**Review by JonathanHallam — APPROVED** [review]
nice, thanks

**Pull request merged**